### PR TITLE
feat(helmfile): oci support

### LINKED ISSUE #13656: Add support for OCI registries in helmfile
### What would you like Renovate to be able to do?

It would be nice to have support to be able to use OCI registries in helmfiles and have renovate be able to update dependencies on that. Currently, renovate only supports using helm chart repositories.

I created a minimum reproduction repo of this here: https://github.com/pspurlock07/renovate-helmfile-oci-reproduction

### If you have any ideas on how this should be implemented, please tell us here.

In a way, similar to how it is being implemented in the helmv3 manager.

Currently, helmfile requires oci to be specified as a key if being used:

```yaml
repositories:
- name: test-app-chart
  url: ghcr.io/pspurlock07
  oci: true
```

Using this example, if `oci:true` then set the `datasource: docker` or something to that effect.

```yaml
releases:
- name: test-app-chart
  namespace: {{ .Values.namespace }}
  chart: test-app-chart/test-app
```

I believe what may be the tricker part is the name that needs be used in the lookupName appears at the end of `chart: test-app-chart/**test-app**`

so an example lookupName that works would be `ghcr.io/pspurlock07/test-app`

The release name does not need to match/can be whatever a user wants. What's important is what they put after the slash ( / ) in their `chart: ` value under releases:

### Is this a feature you are interested in implementing yourself?

No

### PR DESCRIPTION
**Pull request by jycamier**
## Changes

The helmfile manager need to use docker datasource when it deals with OCI repository like : 

```yaml
repositories:
  - name: jycamier
    url: ghcr.io/jycamier/renovate-helmfile-oci-reproduction-app
    oci: true

releases:
  - name: app-test
    chart: jycamier/oci-chart-test
    version: 0.1.0
```

## Context

Close #13656 

## Documentation

- [ ] I have updated the documentation, or
- [x] No documentation update is required

The `oci: true` is a helmfile native attribute.

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

The real repository is https://github.com/jycamier/renovate-helmfile-oci-reproduction.
You can see the pull request here : https://github.com/jycamier/renovate-helmfile-oci-reproduction/pull/2

### CODE REVIEW

**Review by viceice — CHANGES_REQUESTED** [review]
needs tests

**Comment by viceice** [comment]
and git author/ committer fix (use force push to update)

**Comment by jycamier** [comment]
Thank you @viceice 
I think it's good !

File: lib/modules/manager/helmfile/extract.ts [hunk 1]
```diff
@@ -14,6 +15,13 @@ function extractYaml(content: string): string {
   return content.replace(/(^|:)\s*{{.+}}\s*$/gm, '$1');
 }
 
+function getDatasourceId(repos: any, repoName: string): string {
```
Comment (viceice) [inline lib/modules/manager/helmfile/extract.ts:18]: no `any`, please create/ use a proper type

File: lib/modules/manager/helmfile/types.ts [hunk 2]
```diff
@@ -7,5 +7,6 @@ export interface Doc {
   repositories?: {
```
Comment (viceice) [inline lib/modules/manager/helmfile/types.ts:7]: see other comment, maybe you need to extract a new sub type

Comment (jycamier) [reply lib/modules/manager/helmfile/types.ts]: Done !

**Comment by jycamier** [comment]
And my tests are succeed now.

**Review by rarkins — DISMISSED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**